### ISSUE TITLE
clean up the plugin, reduce the global variables

### ISSUE THREAD

**Issue by chrisrhayden** [kevinywlui/zlong_alert.zsh#13]
It would be nice to not pollute the global env so much. I think we should check the variables inside the functions they are used.

We should also clean up some of the small stuff like not quoting variables. I know its not always necessary in zsh but I think it makes some things confusing and has cased issues in the past.

The style should also be consistent. I'm fine with any style just as long as the plugin is consistent.

**Comment by fdelavier** [comment]
Hi @chrisrhayden,

Great idea!

Just one comment :

> It would be nice to not pollute the global env so much. I think we should check the variables inside the functions they are used.

The global env still has an interest : it's executed only 1 time, when the shell is started (not each time a command is executed)

**Comment by chrisrhayden** [comment]
@fdelavier

i agree, at the same time there shouldn't be any noticeable performance impact. we can also exit early with.

```bash
zlong_alert_post() {
    if [[ -z "$zlong_duration" ]]; then
        local zlong_duration=30
    fi

    if [[ -z "$zlong_last_cmd" ]] \
        || ((("$EPOCHSECONDS" - "$zlong_timestamp") <= "$zlong_duration")); then
        echo 'return'
    fi

    # the rest of the function ...
}
```

i have no real reason other then i think it would be nice for people who just use the defaults or just want to interact with one or two. honestly i dont use this plugin anymore, and you have put in more work then i have so ill leave the decision up to you.

**Comment by fdelavier** [comment]
Hi,

I sent a first PR, mainly for adding quotes when It looks to be appropriate (not everywhere, because quotes may sometimes break the script).

I'll try to look how other pluging handle parameter variables.

**Comment by chrisrhayden** [comment]
thanks, i appreciate what you have been doing. i should have more time the end of this week to help.
